### ISSUE TITLE
New-style OOP support

### ISSUE THREAD

**Issue by igurin-invn** [gllmflndn/m2html#5]
Support new-style classes (i.e., using `classdef`). The tool now considers classes to be scripts.

**Comment by igurin-invn** [comment]
I spent hours trying to do this myself, and got nowhere.

**Comment by gllmflndn** [comment]
Sorry about this, m2html was written before `classdef` was introduced. There might be some forks around where its support was added.
A starting point would be to add `classdef` as an alternative to `function` here:
https://github.com/gllmflndn/m2html/blob/main/private/mfileparse.m#L50
Next step involves parsing properties and methods and storing them appropriately in the output `s` structure.

**Comment by igurin-invn** [comment]
I see.

I don't know where else to ask this, so I'll ask here. Are you interested in reviewing changes by others? I made some changes on my side that remove most of the Code Analyzer warnings and should (according to Code Analyzer) improve performance a bit. I tried to push, but GitHub wouldn't allow me to.

**Comment by gllmflndn** [comment]
Yes, you are very much welcome to open a [pull request](https://github.com/gllmflndn/m2html/pulls). It might take me some time to review the changes though.

**Comment by igurin-invn** [comment]
```
git.exe push --progress "origin" IG
remote: Permission to gllmflndn/m2html.git denied to igurin-invn.
fatal: unable to access 'https://github.com/gllmflndn/m2html.git/': The requested URL returned error: 403
```

**Comment by gllmflndn** [comment]
You need to use the _fork -> clone -> branch -> edit -> commit -> push -> pull request_ workflow, see for example:
https://github.com/firstcontributions/first-contributions/blob/master/README.md
See also:
https://guides.github.com/activities/forking/
https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests

**Comment by igurin-invn** [comment]
I see. Trying to follow the workflow, but I can't find `Contributors.md`.
